lib/widgets/dialog.dart

```diff
@@ -15,26 +15,23 @@ Widget _dialogActionText(String text) {
   );
 }
 
+/// Displays an [AlertDialog] with a dismiss button.
+///
+/// Returns a [Future] that resolves when the dialog is closed.
 Future<void> showErrorDialog({
```

Comment:
This behavior (which `showDialog` has and we faithfully wrap) is kind of a bad pattern for an API, as we've discussed in the context of #731 / the `unawaited_futures` lint. If a function is named "show thing" and its return type is some `Future<T>`, the future should complete when the thing has been shown, not when some possibly-much-later event happens.

I think a good model here is `ScaffoldMessengerState.showSnackBar`: there's a Future it wants to provide for callers to potentially await on, but it's for something that happens after the function's own work is done, so instead of returning a value that *is* a Future it returns a value that *has* a Future. And the property name, `closed`, makes it clear what event is being awaited.

Anyway, that's a problem for outside this PR.